src/test/java/com/google/sps/DestinationDataServletTest.java

```diff
@@ -0,0 +1,148 @@
+  private static final String HINT1_PARAMETER = "hint1";
+  private static final String HINT2_PARAMETER = "hint2";
+  private static final String HINT3_PARAMETER = "hint3";
+  private static final String OBSCURITY_PARAMETER = "obscurity";
+  private static final String TAG_PARAMETER = "tag";
+  private static final String PLACEID_PARAMETER = "placeId";
+  private static final String HOME_URL = "/index.html";
+  private static final String NAME_INPUT = "Golden Gate Bridge";
+  private static final String LAT_INPUT = "123.456";
+  private static final String LNG_INPUT = "234.567";
+  private static final String CITY_INPUT = "San Francisco";
+  private static final String DESC_INPUT = "Famous Bridge in SF";
+  private static final String RIDDLE_INPUT = "Stay away from me if you're afraid of heights";
+  private static final String H1_INPUT = "Overlooks the water";
+  private static final String H2_INPUT = "Golden-red in color";
+  private static final String H3_INPUT = "You have to pay to use me";
+  private static final String[] TAG_INPUT = {"historical", "tourist"};
+  private static final String OBSCURITY_INPUT = "easy";
+  private static final String PLACEID_INPUT = "123";
+
+  private static final Gson GSON = new Gson();
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() throws IOException {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void storeInDatastore() throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    servlet = new DestinationDataServlet(datastore);
+    doReturn(NAME_INPUT).when(request).getParameter(NAME_PARAMETER);
+    doReturn(LAT_INPUT).when(request).getParameter(LAT_PARAMETER);
+    doReturn(LNG_INPUT).when(request).getParameter(LNG_PARAMETER);
+    doReturn(CITY_INPUT).when(request).getParameter(CITY_PARAMETER);
+    doReturn(DESC_INPUT).when(request).getParameter(DESCRIPTION_PARAMETER);
+    doReturn(RIDDLE_INPUT).when(request).getParameter(RIDDLE_PARAMETER);
+    doReturn(H1_INPUT).when(request).getParameter(HINT1_PARAMETER);
+    doReturn(H2_INPUT).when(request).getParameter(HINT2_PARAMETER);
+    doReturn(H3_INPUT).when(request).getParameter(HINT3_PARAMETER);
+    doReturn(TAG_INPUT).when(request).getParameterValues(TAG_PARAMETER);
+    doReturn(OBSCURITY_INPUT).when(request).getParameter(OBSCURITY_PARAMETER);
+    doReturn(PLACEID_INPUT).when(request).getParameter(PLACEID_PARAMETER);
+    servlet.doPost(request, response);
+
+    Assert.assertEquals(
+        1, datastore.prepare(new Query(Constants.DESTINATION_ENTITY)).countEntities());
+
+    Query query = new Query(Constants.DESTINATION_ENTITY);
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Destination> destinations = new ArrayList();
```

Comment:
Because this will always have a single destination entity, you should be able to just use `results.asSingleEntity()` instead of building a list.